### PR TITLE
Support to set custom key/value pairs

### PR DESCRIPTION
**Pull request by akshayajeevan**
Added methods for setting custom key-value 

#27

### CODE REVIEW

File: www/FirebaseCrash.js [hunk 1]
```diff
@@ -21,5 +21,21 @@ module.exports = {
...
+        return new Promise(function(resolve, reject) {
+            if (typeof key !== "string") {
+                return reject(new TypeError("Custom key must be a string"));
+            }
+            if (typeof value !== "string" && typeof value !== "number" && typeof value !== "boolean") {
+                return reject(new TypeError("Custom value must be a string/number/boolean"));
+            } else if(typeof value === "string") {
+                exec(resolve, reject, PLUGIN_NAME, "setStringValue", [key, value]);
+            } else if(typeof value === "number") {
+                exec(resolve, reject, PLUGIN_NAME, "setNumValue", [key, value]);
```
Comment (chemerisuk) [inline www/FirebaseCrash.js]: Let's rename the method to be less opinionated.
```suggestion
                exec(resolve, reject, PLUGIN_NAME, "setNumberValue", [key, value]);
```

File: www/FirebaseCrash.js [hunk 2]
```diff
@@ -21,5 +21,21 @@ module.exports = {
...
+            } else if(typeof value === "boolean") {
+                exec(resolve, reject, PLUGIN_NAME, "setBoolValue", [key, value]);
```
Comment (chemerisuk) [inline www/FirebaseCrash.js]: Let's rename the method to be less opinionated.
```suggestion
                exec(resolve, reject, PLUGIN_NAME, "setBooleanValue", [key, value]);
```

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cordova-plugin-firebase-crash",
-  "version": "4.1.2",
+  "version": "4.2.0",
```
Comment (chemerisuk) [inline package.json]: Please rollback version change from PR. I'll publish new release with tag as soon as the PR will me merged.
```suggestion
  "version": "4.1.2",
```

File: plugin.xml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <plugin xmlns="http://apache.org/cordova/ns/plugins/1.0"
 xmlns:android="http://schemas.android.com/apk/res/android"
            id="cordova-plugin-firebase-crash"
-      version="4.1.2">
+      version="4.2.0">
```
Comment (chemerisuk) [inline plugin.xml]: Please rollback version change from PR. I'll publish new release with tag as soon as the PR will me merged.
```suggestion
      version="4.1.2">
```

**Review by chemerisuk — CHANGES_REQUESTED** [review]
Thanks @akshayajeevan, changes are looking good! Just apply couple of small tweaks and we are ready to go.

File: www/FirebaseCrash.js (hunk 2 above)
Comment (akshayajeevan) [reply www/FirebaseCrash.js]: Done

File: www/FirebaseCrash.js (hunk 1 above)
Comment (akshayajeevan) [reply www/FirebaseCrash.js]: Done

File: package.json (hunk 3 above)
Comment (akshayajeevan) [reply package.json]: Done

File: plugin.xml (hunk 4 above)
Comment (akshayajeevan) [reply plugin.xml]: Done

**Comment by chemerisuk** [comment]
@akshayajeevan LGTM 🍺

**Comment by chemerisuk** [comment]
Published in 4.2.0

**Pull request merged**